### PR TITLE
Remove the dependency on async-lock

### PR DESCRIPTION
**Pull request by james7132**
This PR removes the need for async-lock as a dependency by making the Executor initializable in a const context. This relies on the const initialization of CondVars and Mutexes, so the MSRV was bumped to 1.68.

This MSRV bump may not be acceptable, so either this PR must wait, or an alternative queue implementation needs to be chosen.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -280,11 +280,14 @@ impl Executor {
         );
         let _enter = span.enter();
 
+        let thread_limit = inner
+            .thread_limit
+            .get_or_insert_with(|| NonZeroUsize::new(Self::max_threads()).unwrap())
```
Comment (notgull) [inline src/lib.rs]: Rather than having `unwrap` here, I would rather have `max_threads` return `NonZeroUsize`.

**Review by notgull — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (james7132) [reply src/lib.rs]: Done.

**Review by notgull — CHANGES_REQUESTED** [review]
As an organization we maintain an MSRV policy where the MSRV should not exceed Debian Stable's packaged version of rustc. In this case this is 1.63.0. So we need to wait before this can be merged.

**Comment by taiki-e** [comment]
> This relies on the const initialization of CondVars and Mutexes, so the MSRV was bumped to 1.68.

([const Mutex::new](https://doc.rust-lang.org/nightly/std/sync/struct.Mutex.html#method.new) and [const Condvar::new](https://doc.rust-lang.org/nightly/std/sync/struct.Condvar.html#method.new) are available since 1.63. That actually requires 1.68 is [const VecDeque::new](https://doc.rust-lang.org/nightly/std/collections/struct.VecDeque.html#method.new))

**Comment by james7132** [comment]
Found a way to lower the MSRV to 1.63 by using `Option<VecDeque<T>>` instead. This shouldn't need extra branches due to the niche layout optimization of VecDeque and Option.

**Review by notgull — APPROVED** [review]
Thanks!

**Pull request merged**